backend/src/discovita/service/openai/examples/function_calling_example.py

```diff
@@ -0,0 +1,218 @@
+"""
+Function Calling Example with OpenAI Responses API
+
+This example demonstrates how to use the OpenAI Responses API for function calling.
+It shows how to define functions, handle function calls, and process the results.
+
+The example implements a simple calculator that can perform basic arithmetic
+operations (add, subtract, multiply, divide) and a weather service that returns
+weather information for a given location.
+"""
```

Comment:
For consistency, move this under backend/scripts.   That's just my preference for any executable code that we don't plan to run on the server but want in the codebase.